frontend/src/scenes/hog-functions/logs/logsViewerLogic.tsx

```diff
@@ -112,7 +113,7 @@ const loadLogs = async (request: LogEntryParams): Promise<LogEntry[]> => {
         ${hogql.raw(buildBoundaryFilters(request))}
         ${hogql.raw(buildSearchFilters(request))}
         ORDER BY timestamp ${hogql.raw(request.order)}
-        LIMIT ${LOG_VIEWER_LIMIT}`
+        LIMIT ${request.limit ?? LOG_VIEWER_LIMIT}`
```

Comment:
maybe this should be limit and offset so that the request isn't just re-retrieving the same logs on each request along with the new ones? happy for this to go in and get fixed later tho.